src/components/PDFView/index.js

```diff
@@ -45,21 +57,66 @@ class PDFView extends Component {
     }
 
     /**
-     * Upon successful document load, set the number of pages on PDF,
+     * Upon successful document load, combine an array of page viewports,
+     * set the number of pages on PDF,
      * hide/reset PDF password form, and notify parent component that
      * user input is no longer required.
      *
-     * @param {*} {numPages} No of pages in the rendered PDF
+     * @param {*} pdf The PDF file instance
      * @memberof PDFView
      */
-    onDocumentLoadSuccess({numPages}) {
-        this.setState({
-            numPages,
-            shouldRequestPassword: false,
-            isPasswordInvalid: false,
+    onDocumentLoadSuccess(pdf) {
+        const numPages = pdf.numPages;
+
+        Promise.all(
+            _.times(numPages, (index) => {
+                const pageNumber = index + 1;
+
+                return pdf.getPage(pageNumber).then((page) => page.getViewport({scale: 1}));
+            }),
+        ).then((pageViewports) => {
+            this.setState({
+                pageViewports,
+                numPages,
+                shouldRequestPassword: false,
+                isPasswordInvalid: false,
+            });
         });
     }
 
+    /**
+     * Calculates a proper page height.
+     * It is based on a ratio between the specific page viewport width and provided page width.
+     * Also, the app should take into account the page borders.
+     * @param {*} pageIndex
+     * @returns {Number}
+     */
+    calculatePageHeight(pageIndex) {
+        if (this.state.pageViewports.length === 0) {
+            throw new Error('calculatePageHeight() called too early');
```

Comment:
Maybe should we just `return` early here, instead of throwing an error? This gets called when `List` is initiated right? I'm worried this might break the application in an edge-case for no good reason

Reply:
Throwing an `Error` would lead to a crash in the App, wouldn't?

Reply:
Yeah, good catch. 
I am going to return zero and call `Log.warn` with a message. Is it okay?
Commit - https://github.com/Expensify/App/pull/22760/commits/c95c7754f11fafd9ff745a339c99fd2510e681a2

Reply:
Also, actually, this method is being called only when the app already renders pages.
```javascript
{
    this.state.pageViewports.length > 0 && (
        <List
            // Here we go
            estimatedItemSize={this.calculatePageHeight(0)}
            itemSize={this.calculatePageHeight}
        >
            {this.renderPage}
        </List>
    );
}
```